### PR TITLE
Add Semantic Segmentation Interface

### PR DESCRIPTION
**Pull request by iizukak**
## Motivation and Context

This PR implement #88.
You can use Semantic Segmentation from blueoil entry point.

## Description

In this PR, theses features are implemented.

- Enable `semantic_segmentation` task from blueoil entry script.
- Implement `CamvidBase` and `CamvidCustom` classes.
  - Make `Camvid` class inherit `CamvidBase` class.
  - Make `CamvidCustom` class inherit `StoragePathCustomizable` and `CamvidBase`.
  - `StoragePathCustomizable` are required to separate dataset for training and test if user only have training dataset.
- Unit Tests for these classes.
  - This test include very reduced `Camvid` and `CamvidCustom` dataset.
- Write `CamvidCustom` documentation.

### Unit test

I added `test_camvid.py` for unit testing.
In this script, `Camvid` and `CamvidCustom` class's basic functionality are tested.
`CamvidCustom` may be only contain training dataset. `DummyCamvidCustomWithoutTestDataset` is that case.

`_show_image_with_annotation` function are debug helper script. To see original and annotated image, please comment out `#` in these lines.

https://github.com/blue-oil/blueoil/pull/93/files#diff-9fbc1734a50a11ddf3811d724af7508eR71
https://github.com/blue-oil/blueoil/pull/93/files#diff-9fbc1734a50a11ddf3811d724af7508eR83

You can see like this.

### [WIP] Manualy test

I tested `Camvid` and `CamvidCustom`datases with manual experiment.

#### CamvidCustom

- Config file
- Screenshot of tensorboard

#### Camvid

- Config file
- Screenshot of tensorboard

To be written.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: lmnet/lmnet/datasets/camvid.py [hunk 1]
```diff
@@ -60,35 +60,10 @@ def fetch_one_data(args):
...
 
 
-class Camvid(SegmentationBase):
-    """CamVid
-
-    http://www0.cs.ucl.ac.uk/staff/G.Brostow/papers/Brostow_2009-PRL.pdf
-    https://github.com/alexgkendall/SegNet-Tutorial/tree/master/CamVid
+class CamvidBase(SegmentationBase):
+    """
+    Base class for CaVid and the variant dataset formats.
```
Comment (nlpng) [inline lmnet/lmnet/datasets/camvid.py]: 'CaVid' -> 'CamVid'

Comment (iizukak) [reply lmnet/lmnet/datasets/camvid.py]: Thanks. Fixed.

**Comment by iizukak** [comment]
Oh, I missed `blueoil_test.sh` to fix.

File: docs/usage/dataset.md [hunk 2]
```diff
@@ -126,3 +129,81 @@ Example of `class-descriptions.csv`
 
 ### DeLTA-Mark object detection
 You can download data of this format by using [Delta-Mark](https://delta.leapmind.io/mark/) service.
+
+
+### CamvidCustom
+
+It is data format based on [CamVid](http://mi.eng.cam.ac.uk/research/projects/VideoRec/CamVid/). We can see sample dataset from [SegNet-Turorial](https://github.com/alexgkendall/SegNet-Tutorial) repository. In CamvidCustom dataset format, training and annotation data are both binary image file.
```
Comment (nlpng) [inline docs/usage/dataset.md]: `SegNet-Turorial` -> `SegNet-Tutorial`

File: docs/usage/dataset.md [hunk 3]
```diff
@@ -126,3 +129,81 @@ Example of `class-descriptions.csv`
...
+
+Place the following files and directory.
+
+- `label_colors.txt`: Class information file like under the example.
+- `train.txt`: The space separated pairs of data file path and annotation file path like under the example.
```
Comment (nlpng) [inline docs/usage/dataset.md]: Sorry, I didn't quite get the meaning of this description compare to the example given below.

File: docs/usage/dataset.md (hunk 2 above)
Comment (iizukak) [reply docs/usage/dataset.md]: thanks. fixed.

File: docs/usage/dataset.md (hunk 3 above)
Comment (iizukak) [reply docs/usage/dataset.md]: OK, I'm thinking better description...

**Comment by iizukak** [comment]
@ruimashita Please review. sorry for big diff...

**Comment by ruimashita** [comment]
@iizukak Thanks. Could you separate the PR into several small PRs?

**Comment by iizukak** [comment]
@ruimashita Ok.

**Pull request closed without merging**